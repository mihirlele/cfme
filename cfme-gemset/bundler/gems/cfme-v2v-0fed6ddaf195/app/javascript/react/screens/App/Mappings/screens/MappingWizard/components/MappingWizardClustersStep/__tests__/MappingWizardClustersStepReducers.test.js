import Immutable from 'seamless-immutable';
import clustersReducer, { initialState } from '../MappingWizardClustersStepReducer';
import { FETCH_V2V_SOURCE_CLUSTERS, QUERY_V2V_PROVIDERS } from '../MappingWizardClustersStepConstants';
import { sourceClusters } from '../mappingWizardClustersStep.fixtures';

test('sets default state', () => {
  const action = { type: '@@INIT' };
  const state = clustersReducer(undefined, action);

  expect(state).toMatchSnapshot();
});

describe('fetching source clusters', () => {
  test('is pending', () => {
    const action = {
      type: `${FETCH_V2V_SOURCE_CLUSTERS}_PENDING`
    };
    const prevState = initialState.set('isRejectedSourceClusters', true);
    const state = clustersReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });

  test('is rejected', () => {
    const action = {
      type: `${FETCH_V2V_SOURCE_CLUSTERS}_REJECTED`,
      payload: 'error'
    };
    const prevState = initialState.set('isFetchingSourceClusters', true);
    const state = clustersReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });

  describe('is successful,', () => {
    let prevState;
    beforeEach(() => {
      prevState = initialState.set('isRejectedSourceClusters', true).set('isFetchingSourceClusters', true);
    });

    test('and there is data', () => {
      const { resources } = sourceClusters;
      const payload = {
        data: {
          ...sourceClusters,
          resources: [
            {
              ...resources
            }
          ]
        }
      };
      const action = {
        type: `${FETCH_V2V_SOURCE_CLUSTERS}_FULFILLED`,
        payload
      };
      const state = clustersReducer(prevState, action);

      expect(state).toMatchSnapshot();
    });

    test('and there is no data', () => {
      const payload = {
        data: {
          ...sourceClusters,
          resources: []
        }
      };
      const action = {
        type: `${FETCH_V2V_SOURCE_CLUSTERS}_FULFILLED`,
        payload
      };
      const state = clustersReducer(prevState, action);

      expect(state).toMatchSnapshot();
    });
  });
});

describe('QUERY_V2V_PROVIDERS', () => {
  test('querying providers is pending', () => {
    const action = { type: `${QUERY_V2V_PROVIDERS}_PENDING` };
    const prevState = Immutable({
      ...initialState,
      isRejectedQueryProviders: true
    });
    const state = clustersReducer(prevState, action);

    expect(state).toEqual({
      ...prevState,
      isQueryingProviders: true,
      isRejectedQueryProviders: false
    });
  });

  test('querying providers is fulfilled', () => {
    const payload = {
      data: {
        results: [{ mock: 'data' }]
      }
    };
    const action = {
      type: `${QUERY_V2V_PROVIDERS}_FULFILLED`,
      payload
    };
    const prevState = Immutable({
      ...initialState,
      isQueryingProviders: true,
      isRejectedQueryProviders: true,
      errorQueryProviders: 'error'
    });
    const state = clustersReducer(prevState, action);

    expect(state).toEqual({
      ...prevState,
      providers: payload.data.results,
      isQueryingProviders: false,
      isRejectedQueryProviders: false,
      errorQueryProviders: null
    });
  });

  test('fetching a plan is rejected', () => {
    const payload = { error: 'error' };
    const action = {
      type: `${QUERY_V2V_PROVIDERS}_REJECTED`,
      payload
    };
    const prevState = Immutable({
      ...initialState,
      isQueryingProviders: true
    });
    const state = clustersReducer(prevState, action);

    expect(state).toEqual({
      ...prevState,
      isQueryingProviders: false,
      isRejectedQueryProviders: true,
      errorQueryProviders: payload
    });
  });
});
