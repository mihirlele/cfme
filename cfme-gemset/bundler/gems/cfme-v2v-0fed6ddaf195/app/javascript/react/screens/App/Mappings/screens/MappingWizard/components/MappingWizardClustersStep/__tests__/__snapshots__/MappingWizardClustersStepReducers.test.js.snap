// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetching source clusters is pending 1`] = `
Object {
  "errorQueryProviders": null,
  "errorSourceClusters": null,
  "errorTargetClusters": null,
  "isFetchingSourceClusters": true,
  "isFetchingTargetClusters": false,
  "isQueryingProviders": false,
  "isRejectedQueryProviders": false,
  "isRejectedSourceClusters": false,
  "isRejectedTargetClusters": false,
  "providers": Array [],
  "sourceClusters": Array [],
  "targetClusters": Array [],
}
`;

exports[`fetching source clusters is rejected 1`] = `
Object {
  "errorQueryProviders": null,
  "errorSourceClusters": "error",
  "errorTargetClusters": null,
  "isFetchingSourceClusters": false,
  "isFetchingTargetClusters": false,
  "isQueryingProviders": false,
  "isRejectedQueryProviders": false,
  "isRejectedSourceClusters": true,
  "isRejectedTargetClusters": false,
  "providers": Array [],
  "sourceClusters": Array [],
  "targetClusters": Array [],
}
`;

exports[`fetching source clusters is successful, and there is data 1`] = `
Object {
  "errorQueryProviders": null,
  "errorSourceClusters": null,
  "errorTargetClusters": null,
  "isFetchingSourceClusters": false,
  "isFetchingTargetClusters": false,
  "isQueryingProviders": false,
  "isRejectedQueryProviders": false,
  "isRejectedSourceClusters": false,
  "isRejectedTargetClusters": false,
  "providers": Array [],
  "sourceClusters": Array [
    Object {
      "0": Object {
        "created_on": "2016-07-15T18:28:16Z",
        "drs_automation_level": null,
        "drs_enabled": null,
        "drs_migration_threshold": null,
        "effective_cpu": null,
        "effective_memory": null,
        "ems_id": "10000000000012",
        "ems_ref": "/api/clusters/00000002-0002-0002-0002-00000000024b",
        "ems_ref_obj": "/api/clusters/00000002-0002-0002-0002-00000000024b",
        "ext_management_system": Object {
          "emstype": "vmwarews",
        },
        "ha_admit_control": null,
        "ha_enabled": null,
        "ha_max_failures": null,
        "href": "http://localhost:3000/api/clusters/10000000000001",
        "id": "1",
        "last_perf_capture_on": null,
        "name": "VMWare Cluster1",
        "type": null,
        "uid_ems": "00000002-0002-0002-0002-00000000024b",
        "updated_on": "2016-07-15T18:28:16Z",
        "v_parent_datacenter": "Datacenter",
      },
      "1": Object {
        "created_on": "2016-07-15T18:28:16Z",
        "drs_automation_level": null,
        "drs_enabled": null,
        "drs_migration_threshold": null,
        "effective_cpu": null,
        "effective_memory": null,
        "ems_id": "10000000000012",
        "ems_ref": "/api/clusters/00000002-0002-0002-0002-00000000024b",
        "ems_ref_obj": "/api/clusters/00000002-0002-0002-0002-00000000024b",
        "ext_management_system": Object {
          "emstype": "vmwarews",
        },
        "ha_admit_control": null,
        "ha_enabled": null,
        "ha_max_failures": null,
        "href": "http://localhost:3000/api/clusters/10000000000002",
        "id": "2",
        "last_perf_capture_on": null,
        "name": "VMWare Cluster2",
        "type": null,
        "uid_ems": "00000002-0002-0002-0002-00000000024b",
        "updated_on": "2016-07-15T18:28:16Z",
        "v_parent_datacenter": "Datacenter",
      },
      "2": Object {
        "created_on": "2016-07-15T18:28:16Z",
        "drs_automation_level": null,
        "drs_enabled": null,
        "drs_migration_threshold": null,
        "effective_cpu": null,
        "effective_memory": null,
        "ems_id": "10000000000012",
        "ems_ref": "/api/clusters/00000002-0002-0002-0002-00000000024b",
        "ems_ref_obj": "/api/clusters/00000002-0002-0002-0002-00000000024b",
        "ext_management_system": Object {
          "emstype": "vmwarews",
        },
        "ha_admit_control": null,
        "ha_enabled": null,
        "ha_max_failures": null,
        "href": "http://localhost:3000/api/clusters/10000000000003",
        "id": "3",
        "last_perf_capture_on": null,
        "name": "VMWare Cluster3",
        "type": null,
        "uid_ems": "00000002-0002-0002-0002-00000000024b",
        "updated_on": "2016-07-15T18:28:16Z",
        "v_parent_datacenter": "Datacenter",
      },
    },
  ],
  "targetClusters": Array [],
}
`;

exports[`fetching source clusters is successful, and there is no data 1`] = `
Object {
  "errorQueryProviders": null,
  "errorSourceClusters": null,
  "errorTargetClusters": null,
  "isFetchingSourceClusters": false,
  "isFetchingTargetClusters": false,
  "isQueryingProviders": false,
  "isRejectedQueryProviders": false,
  "isRejectedSourceClusters": false,
  "isRejectedTargetClusters": false,
  "providers": Array [],
  "sourceClusters": Array [],
  "targetClusters": Array [],
}
`;

exports[`sets default state 1`] = `
Object {
  "errorQueryProviders": null,
  "errorSourceClusters": null,
  "errorTargetClusters": null,
  "isFetchingSourceClusters": false,
  "isFetchingTargetClusters": false,
  "isQueryingProviders": false,
  "isRejectedQueryProviders": false,
  "isRejectedSourceClusters": false,
  "isRejectedTargetClusters": false,
  "providers": Array [],
  "sourceClusters": Array [],
  "targetClusters": Array [],
}
`;
